[quant][api] Add assert for backend in get_default_qconfig related apis

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86259

Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

jerryzh168 committed:
[quant][api] Add assert for backend in get_default_qconfig related apis

Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

andrewor14 commented on torch/ao/quantization/qconfig.py:
```diff
@@ -228,12 +228,19 @@ def get_default_qconfig(backend='x86', version=0):
     """
+    supported_backends = ["fbgemm", "x86", "qnnpack", "onednn"]
+    if backend not in supported_backends:
+        raise AssertionError(
```
Should this be `ValueError` instead?

jerryzh168 replied:
I actually checked this as well, ValueError means the type is correct and value is not correct, but in this case the type may not be correct either so I guess AssertionError is OK. we are also using AssertionError for the version number as well

jerryzh168 commented:
@pytorchbot merge -g

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge

jerryzh168 commented:
@pytorchbot merge -g

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge -g

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 commented:
@pytorchbot merge

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 commented:
@pytorchbot merge -g

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][api] Add assert for backend in get_default_qconfig related apis"


Summary:
Add assertion to make sure backend is one of "fbgemm", "x86", "qnnpack" and "onednn"
for get_default_qconfig, get_default_qat_qconfig, get_default_qconfig_mapping and get_default_qat_qconfig_mapping

Test Plan:
python test/test_quantization.py -k test_get_default_qconfig_mapping

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40236474](https://our.internmc.facebook.com/intern/diff/D40236474)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40236474).

jerryzh168 commented:
closed in favor of https://github.com/pytorch/pytorch/pull/87331
